core/interfaces/i_deletearea.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview The interface for a component that can delete a block that is
+ *   dropped on top of it.
+ * @author aschmiedt@google.com (Abby Schmiedt)
+ */
+
+'use strict';
+
+goog.provide('Blockly.IDeleteArea');
+
+
+/** @interface */
+Blockly.IDeleteArea = function() {};
```

Comment:
Not necessarily in this PR, but I imagine the goal here would be to turn `Blockly.WorkspaceSvg.prototype.recordDeleteAreas` and `Blockly.WorkspaceSvg.prototype.isDeleteArea` into an API that supports ANY delete area where areas get "registered" onto the workspace during startup, and these two methods just iterate through the list of delete areas completing the action at hand.